app/uk/gov/hmrc/customs/notification/controllers/CustomsNotificationBlockedController.scala

```diff
@@ -57,7 +57,34 @@ class CustomsNotificationBlockedController @Inject()(val logger: NotificationLog
       }
   }
 
-  private def response(count: Int): Result = {
+  def deleteBlocked():Action[AnyContent] = Action.async {
+    implicit request =>
+      request.headers.get(X_CLIENT_ID_HEADER_NAME).fold {
+        logger.errorWithHeaders(s"missing $X_CLIENT_ID_HEADER_NAME header when calling blocked-flag endpoint", request.headers.headers)
+        Future.successful(ErrorClientIdMissing.XmlResult)
+      } { clientId =>
```

Comment:
I think `new HasId {...` could be extracted to a private function as well

Reply:
Done